library/src-bootstrapped/scala/tests/TypeChecking.scala

```diff
@@ -0,0 +1,15 @@
+package scala.tests
+
+import scala.quoted._
+import scala.tasty.Reflection
+
+object TypeChecking {
+
+  inline def typeChecks(inline code: String): Boolean = ${ typeChecksImpl(code) }
+
+  private def typeChecksImpl(code: String)(implicit reflect: Reflection): Expr[Boolean] = {
+    import reflect._
+    typing.typeChecks(code).toExpr
+  }
```

Comment:
Maybe remove `private`, as inlining will create an accessor if it's private.

Can we use top-level definitions to get rid of `object TypeChecking`?

Regarding `scala.tests`, what about `scala.test`?

Reply:
I will try the top level definition and change to `scala.test`. `typeChecksImpl` must be private or it would be visible under `import scala.test`.

Reply:
In the future, we will also support arbitrary code in top-level splices.  Which will allow us to implement it like

```scala
inline def typeChecks(inline code: String): Boolean = ${
  val reflect = implicitly[Reflection]
  import reflect._
  typing.typeChecks(code).toExpr
}
```